### PR TITLE
fix novadocker to work with packaging

### PR DESCRIPTION
**Pull request by paulczar**

### CODE REVIEW

File: roles/docker/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+dependencies:
+  - role: apt-repos
+    repos:
+    - repo: 'deb {{ apt_repos.docker.repo }} docker main'
+      key_url: '{{ apt_repos.docker.key_url }}'
```
Comment (j2sol) [inline roles/docker/meta/main.yml]: Looks like some whitespace here was missed.

File: roles/nova-data/tasks/docker.yml [hunk 2]
```diff
@@ -14,7 +12,28 @@
...
   delay: 60
+
+- name: pip install novadocker
+  pip: name='{{ nova.driver.docker.dest }}'
+       extra_args='-i {{ openstack.pypi_mirror }}'
+  register: result
+  until: result|success
+  retries: 3
+  delay: 60
+  when: result|success and openstack_install_method == 'source'
```
Comment (j2sol) [inline roles/nova-data/tasks/docker.yml]: where is this result coming from in this test? Seems weird to have a conditional like this, but maybe I'm missing context.

File: roles/nova-data/tasks/docker.yml [hunk 3]
```diff
@@ -14,7 +12,28 @@
...
+
+- name: pip install novadocker
+  pip: name='{{ nova.driver.docker.dest }}'
+       extra_args='-i {{ openstack.pypi_mirror }}'
+       virtualenv='/opt/bbc/openstack-{{ openstack_package_version }}/nova/'
+  register: result
+  until: result|success
+  retries: 3
+  delay: 60
+  when: result|success and openstack_install_method == 'package'
```
Comment (j2sol) [inline roles/nova-data/tasks/docker.yml]: Same question here.

File: roles/nova-data/tasks/docker.yml [hunk 4]
```diff
@@ -10,11 +8,32 @@
...
+  delay: 60
+
+- name: pip install novadocker
+  pip: name='{{ nova.driver.docker.dest }}'
+       extra_args='-i {{ openstack.pypi_mirror }}'
+  register: result
+  until: result|success
+  retries: 3
+  delay: 60
+  when: git_novadocker|success and openstack_install_method == 'source'
```
Comment (j2sol) [inline roles/nova-data/tasks/docker.yml]: If git_novadocker is not successful, ansible will error the host and the pip task(s) would never be reached. I think the first part of these when statements are unnecessary.

Comment (paulczar) [inline roles/nova-data/tasks/docker.yml]: Ahh yes,  I meant to use |changed for the result to ensure it only runs when git actually does something.   pushed a commit to fix.

**Comment by j2sol** [comment]
Looks fine from my naïve view of docker/vagrant.

**Pull request merged**